### ISSUE TITLE
Gateway timeouts when uploading CSV files

### ISSUE THREAD

**Issue by SRGDamia1** [ODM2/ODM2DataSharingPortal#346]
I'm not sure if it's timing out because it takes too long to process the file or if it's another issue.

The only visible result to the user is the "failed to upload data" pop-up.  I know it's timing out from the developer console on Firefox.

**Comment by SRGDamia1** [comment]
The time-out seems to be at right around a minute.

I'm getting this even with some pretty small files - less than two weeks of 5 minute data for 7 variables.

**Comment by jcaraballo17** [comment]
@SRGDamia1 Sounds like web service is looping infinitely and the web server is killing the request after one minute. It would be very helpful to debug this with an actual datalogger file. Can you provide one of the files you used?

**Comment by horsburgh** [comment]
@SRGDamia1 - are the files too big to email, or could you just send us one of the ones that was timing out?

**Comment by SRGDamia1** [comment]
Sorry.  I need to get my VPN working so I can grab a file from the server at Stroud.  I'm home now for Christmas.

**Comment by SRGDamia1** [comment]
Oops.  Sorry, I'd forgotten about this.  Here are some files.  The first file failed, but succeeded only after I broke it into three files.
[SL082LOG11Dec18_forUpload.zip](https://github.com/ODM2/ODM2DataSharingPortal/files/2738448/SL082LOG11Dec18_forUpload.zip)

**Comment by SRGDamia1** [comment]
Is there any chance this will be looked at?

**Comment by SRGDamia1** [comment]
@horsburgh - yes, I was hoping there might be something on Limnotech's side.  @aufdenkampe ?

**Comment by aufdenkampe** [comment]
This is a troubling bug that seems to have been around for well over month before the transition to LimnoTech's servers, and probably connected to the push of the 0.9.5 release to USU's production servers, given that a good number of CSV-related bugs were worked on for that release. See https://github.com/ODM2/ODM2DataSharingPortal/milestone/12?closed=1.

Given all this, it would really helpful if we might be able to engage @jcaraballo17 a bit to fix this, along with the seemingly related bug reported by #364.

**Comment by horsburgh** [comment]
@aufdenkampe - sorry, we don't have any current funding to work on this project anymore. @jcaraballo17 may have some ideas about how to fix it, or what might have caused it, but I've moved him onto other funded projects.

**Comment by aufdenkampe** [comment]
From @dbressler75 via email on July 9:

> So we've been testing the upload function and found that it can accept up to 5000 rows of data (i.e., about two weeks worth of data points, collected every five minutes).  The issue we're having is that we have files that have a lot more data than that, so in order to upload all of these we would need to chunk the files down to 5000 then upload individually (see attached).  We can prioritize sites and we have an intern that can work on this but ideally MonMW could accept much larger files.  So before we started on this we wanted to see if you/Limnotech was planning to update this feature anytime soon?  
>  
> Sara was considering writing some code to automate the process as well.  We'll adjust as needed if no plans are in place to update MonMW in the near term, but we just wanted to be sure before we proceeded.

**Comment by aufdenkampe** [comment]
Closing this because it is a duplicate of #364.

Please move comments and conversation to #364.
